src/IssueViz/Editor/ErrorTagging/ErrorTagTooltipProvider.cs

```diff
@@ -44,14 +45,19 @@ internal class ErrorTagTooltipProvider : IErrorTagTooltipProvider
     {
         private readonly IVsThemeColorProvider vsThemeColorProvider;
         private readonly INavigateToRuleDescriptionCommand navigateToRuleDescriptionCommand;
+        private readonly ILogger logger;
 
         [ImportingConstructor]
-        public ErrorTagTooltipProvider(IVsThemeColorProvider vsThemeColorProvider, INavigateToRuleDescriptionCommand navigateToRuleDescriptionCommand)
+
+        public ErrorTagTooltipProvider(IVsThemeColorProvider vsThemeColorProvider, INavigateToRuleDescriptionCommand navigateToRuleDescriptionCommand, ILogger logger)
         {
             this.vsThemeColorProvider = vsThemeColorProvider;
             this.navigateToRuleDescriptionCommand = navigateToRuleDescriptionCommand;
+            this.logger = logger;
         }
 
+        private static long instanceCount = 0;
```

Comment:
Wouldn't this number will always grow and could be misleading especially if a project was open for a long time?

Reply:
It will always grow, yes. The difference between "normal" behaviour and "problem" is orders of magnitude, so it will allow us to tell the difference between them. If we are creating thousands of tooltips, we have a problem.

Reply:
As a bonus, the recent tweaks to the verbose logging mean this also gives an easy way to see which thread the code is running on.